go/obscuronode/enclave/types.go

```diff
@@ -1,29 +1,82 @@
 package enclave
 
 import (
+	"crypto/rand"
+	"math"
+	"math/big"
 	"sync/atomic"
 
+	"github.com/ethereum/go-ethereum/rlp"
+
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
 
 	common2 "github.com/obscuronet/obscuro-playground/go/common"
 	oc "github.com/obscuronet/obscuro-playground/go/obscuronode/common"
 )
 
-// Transfers and Withdrawals for now
-type L2TxType uint64
+// L2TxType indicates the type of L2 transaction - either a transfer or a withdrawal for now
+type L2TxType uint8
 
 const (
 	TransferTx L2TxType = iota
 	WithdrawalTx
 )
 
-// L2Tx Only in clear inside the enclave
-type L2Tx struct {
-	ID     common2.TxHash
-	TxType L2TxType
-	Amount uint64
+// L2TxData is the Obscuro transaction data that will be stored encoded in the types.Transaction data field.
+type L2TxData struct {
+	Type   L2TxType
 	From   common.Address
 	To     common.Address
+	Amount uint64
+}
+
+type L2Tx = types.Transaction
+
+// NewL2Transfer creates an L2Tx of type TransferTx.
+func NewL2Transfer(from common.Address, dest common.Address, amount uint64) *L2Tx {
```

Comment:
these functions do not belong here conceptually.
They belong in the client or wallet. basically this function but which will return an "EncryptedL2Tx"